src/devices/Bmp280/Bmp280.cs

```diff
@@ -0,0 +1,455 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+//ported from https://github.com/adafruit/Adafruit_BMP280_Library/blob/master/Adafruit_BMP280.cpp
+//formulas and code examples can also be found in the datasheet http://www.adafruit.com/datasheets/BST-BMP280-DS001-11.pdf
+
+using System;
+using System.Device.I2c;
+using System.Threading.Tasks;
+
+namespace Iot.Device.Bmp280
+{
+    public class Bmp280 : IDisposable
+    {
+
+        private const byte Signature = 0x58;
+
+        private I2cDevice _i2cDevice;
+        private readonly CommunicationProtocol _communicationProtocol;
+        private bool _initialized = false;
+        private readonly CalibrationData _calibrationData;
+        /// <summary>
+        /// The variable _temperatureFine carries a fine resolution temperature value over to the
+        /// pressure compensation formula and could be implemented as a global variable.
+        /// </summary>
+        private int _temperatureFine;
+
+        private enum CommunicationProtocol
+        {
+            I2c
+        }
+
+        public Bmp280(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice;
+            _calibrationData = new CalibrationData();
+            _communicationProtocol = CommunicationProtocol.I2c;
+        }
+
+        private void Begin()
+        {
+            _i2cDevice.WriteByte((byte)Registers.REGISTER_CHIPID);
+            byte readSignature = _i2cDevice.ReadByte();
+
+            if (readSignature != Signature)
+            {
+                return;
+            }
+            _initialized = true;
+
+            //Read the coefficients table
+            ReadCoefficients();
+        }
+
+        /// <summary>
+        ///  Reads the factory-set coefficients 
+        /// </summary>
+        private void ReadCoefficients()
+        {
+            // Read temperature calibration data
+            _calibrationData.DigT1 = Read16FromRegister((byte)Registers.REGISTER_DIG_T1);
```

Comment:
By reading the datasheet, shouldn't some of these be unsigned and some of them signed? I also think that some of them should be interpreted as little endian and some as big endian but I'm not 100% sure.

Reply:
I got data that seemed reasonable when I tested but I'll double check.

Reply:
I only mentioned this because we used to have a very similar binding for a bme280 here: https://github.com/dotnet/corefxlab/blob/master/samples/System.Devices.Gpio.Samples/PressureTemperatureHumiditySensor.cs  and in there I remember that we needed to take into account that some were signed, and some were not.

Reply:
Only the P1 and T1 are unsigned so it seems to be correct. Changed from int/uint to short instead.